cdflow_commands/plugins/ecs.py

```diff
@@ -97,9 +98,15 @@ def _deploy_factory():
         )
         s3_bucket_factory = S3BucketFactory(boto_session, account_id)
         s3_bucket = s3_bucket_factory.get_bucket_name()
-        write_terragrunt_config(
-            metadata.aws_region, s3_bucket, environment_name, component_name
+
+        lock_table_factory = LockTableFactory(boto_session)
```

Comment:
I've got two comments around this bit;

1) `lock_table_factory` as variable name sounds weird to me, would it not be more natural to just call it `terraform_lock`?  The Class name sounds fine as there's also a good split between LockTable and S3 bucket for state.
2) Having that in `build_ecs_plugin` means we need to duplicate this in `infrastructure` plugin, too?  Perhaps it'd be better to acquire a lock before those things are happening?